### PR TITLE
Update certs package and necessary dependency cleanup

### PR DESCRIPTION
**Pull request by stone-z**
`certs` v2 has some changes which are needed to properly create certConfigs here. To use it, some additional dependency pins and refactoring are necessary.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func mainE() error {
 	{
 		newLogger, err = micrologger.New(micrologger.Config{})
 		if err != nil {
-			return microerror.Maskf(err, "micrologger.New")
```
Comment (stone-z) [inline main.go:40]: Are these an important loss?

**Comment by stone-z** [comment]
Will test this before merging

File: go.mod [hunk 2]
```diff
@@ -3,69 +3,52 @@ module github.com/giantswarm/cluster-operator
...
+	github.com/giantswarm/microerror v0.2.0
+	github.com/giantswarm/microkit v0.2.0
+	github.com/giantswarm/micrologger v0.3.1
+	github.com/giantswarm/operatorkit v0.2.1
 	github.com/giantswarm/resource v0.0.0-20201203121554-9ca9ddbde7cf
-	github.com/giantswarm/tenantcluster v0.0.0-20191209135534-800974a4d4bf
-	github.com/giantswarm/to v0.2.0 // indirect
-	github.com/giantswarm/versionbundle v0.0.0-20200203095303-cd94540b7d5a
-	github.com/go-logfmt/logfmt v0.5.0 // indirect
+	github.com/giantswarm/tenantcluster/v2 v2.0.0
```
Comment (MarcelMue) [inline go.mod:22]: This one seems wacky. The changelog in the project is broken so the included changes are unclear to me.

Comment (stone-z) [reply go.mod:22]: Yeah, I looked at the diff and it looks to be entirely dependency bumps similar to this one. Unfortunately neither the release before nor the one after will work

**Comment by stone-z** [comment]
Tested on `gremlin`, -- this actually works better than the previous version. New `certs` seems to have solved the issue with creating kubeconfigs as expected

**Review by MarcelMue — APPROVED** [review]

**Comment by MarcelMue** [comment]
If it works, it works.

**Pull request merged**